semgrep/tests/e2e/snapshots/test_output/test_sarif_output_with_nosemgrep_and_error/results.sarif

```diff
@@ -75,6 +75,23 @@
                 "text": "useless comparison operation `$X == $X` or `$X != $X`; possible bug?"
               }
             },
+            {
```

Comment:
Unclear whether this should appear twice. Seems to be same rule, different path. Maybe I need to fix some deduping logic?

Reply:
I think this is fine. I think before we were adding only rules that found results but I think it's more correct anyway to have all rules run in the sarif output

Reply:
Probably add to changelog to that effect